### PR TITLE
check for Win32 and either MSVC or cross-buildind

### PR DESCRIPTION
**Pull request by fabiangreffrath**
I think this covers all the cases that we officially, i.e. per
README.md, support:

  * Build for Windows
    * Native
      * MSVC
        - Download libraries
      * MinGW
        - Do not download libraries
    * Cross-build
      - Download libraries
  * Build for anything else
    - Do not download libraries

### CODE REVIEW

File: Source/CMakeLists.txt [hunk 1]
```diff
@@ -215,7 +215,7 @@ endif()
 install(FILES ${WOOF_DLLS} DESTINATION .)
 
 # Try get dependencies in MSYS2 environment.
-if(MSYS)
+if(WIN32 AND NOT SDL2_DIR)
```
Comment (rfomin) [inline Source/CMakeLists.txt:218]: I think it doesn't work anywhere except MSYS2.

**Comment by rfomin** [comment]
> MinGW
> * Do not download libraries

I thought the most popular way to run MinGW on Windows is MSYS2.

File: Source/CMakeLists.txt (hunk 1 above)
Comment (fabiangreffrath) [reply Source/CMakeLists.txt:218]: Well, this logic works on Linux, on Linux when cross-building for Windows and in my native Windows shell using MingW.

**Comment by fabiangreffrath** [comment]
> I thought the most popular way to run MinGW on Windows is MSYS2.

Well, apparently my MSYS2 shell on Windows isn't properly recognized as "MSYS" - and it should work for me, right? :wink:

File: Source/CMakeLists.txt (hunk 1 above)
Comment (fabiangreffrath) [reply Source/CMakeLists.txt:218]: > native Windows shell using MingW.

native MSYS2 shell using MinGW

**Comment by rfomin** [comment]
What CMake generator are you using? If you run cmake -G "MSYS Makefiles" it should work.

**Comment by fabiangreffrath** [comment]
I use "UNIX Makefiles", just as our `.travis.sh` script. System detection based on this choice is too fragile to me.

**Comment by rfomin** [comment]
Well, this is broke `cpack -g zip` with MSVC for me :smile:

**Comment by fabiangreffrath** [comment]
> Well, this is broke `cpack -g zip` with MSVC for me 😄

😡 Did the second commit fix it?

**Comment by rfomin** [comment]
It works, thanks.

**Comment by fabiangreffrath** [comment]
I think I'v had enough of CMake for the rest of this year (at least!).

**Pull request merged**